analysis_service/base/models.py

```diff
@@ -0,0 +1,206 @@
+from datetime import datetime, timedelta
+from pytz import UTC
+from django.db import models
+from django.contrib.auth.models import User
+
+from analysis_service.base.util import provisioning, scheduling
+
+
+class Cluster(models.Model):
```

Comment:
It would be great if you could identify eventual unique and unique-together keys so that we can enforce them as db constraints